Modules/faulthandler.c

```diff
@@ -212,6 +216,79 @@ faulthandler_dump_traceback(int fd, int all_threads,
     reentrant = 0;
 }
 
+#ifdef HAVE_EXECINFO_H
+static void
+faulthandler_stack_dump_impl(int fd)
+{
+#define BACKTRACE_SIZE 16
+#define TRACEBACK_ENTRY_MAX_SIZE 256
+    void *callstack[BACKTRACE_SIZE];
+    int frames = backtrace(callstack, BACKTRACE_SIZE);
+    char **strings = backtrace_symbols(callstack, BACKTRACE_SIZE);
+    if (strings == NULL) {
+        PUTS(fd, "  <failed to get stack trace>\n");
+    }
+    else {
+        for (int i = 0; i < frames; ++i) {
+            char entry_str[TRACEBACK_ENTRY_MAX_SIZE];
+            snprintf(entry_str, TRACEBACK_ENTRY_MAX_SIZE, "  %s\n", strings[i]);
+            size_t length = strlen(entry_str) + 1;
+            if (length == TRACEBACK_ENTRY_MAX_SIZE) {
+                /* We exceeded the size, make it look prettier */
+                // Add ellipsis to last 3 characters
+                entry_str[TRACEBACK_ENTRY_MAX_SIZE - 5] = '.';
+                entry_str[TRACEBACK_ENTRY_MAX_SIZE - 4] = '.';
+                entry_str[TRACEBACK_ENTRY_MAX_SIZE - 3] = '.';
+                // Ensure trailing newline
+                entry_str[TRACEBACK_ENTRY_MAX_SIZE - 2] = '\n';
+                // Ensure that it's null-terminated
+                entry_str[TRACEBACK_ENTRY_MAX_SIZE - 1] = '\0';
+            }
+            _Py_write_noraise(fd, entry_str, length);
+        }
+
+        if (frames == BACKTRACE_SIZE) {
+            PUTS(fd, "  <truncated rest of calls>\n");
+        }
+
+        free(strings);
+    }
+#undef BACKTRACE_SIZE
+#undef TRACEBACK_ENTRY_MAX_SIZE
+}
+#else
+static void
+faulthandler_stack_dump_impl(int fd)
+{
+    PUTS(fd, "  <cannot get C stack on this system>\n");
+}
```

Comment:
could make this error message more informative, mentioning glibc and/or execinfo.h

edit: although bsd makes this more complicated... smh

Reply:
I don't think it's worth it. `faulthandler` is just the first step to debugging a crash, nobody is going to try and switch systems to enable the C stack (they'll just use valgrind or something like that).

Reply:
In addition, switching system may also hide the failure (the issue could come from the platform they are working on)